src/System.Management.Automation/engine/NativeCommandParameterBinder.cs

```diff
@@ -162,8 +162,42 @@ private void appendOneNativeArgument(ExecutionContext context, object obj, char
                     arg = PSObject.ToStringParser(context, ParserOps.Current(null, list));
                 }
 
+#if UNIX
+                Collection<string> resolvedPaths = new Collection<string>();
+#endif
                 if (!String.IsNullOrEmpty(arg))
                 {
+                    // This implements globbing of arguments given to native tools on UNIX.
+                    // Conversely, Windows tools implement their own globbing,
```

Comment:
Yes, and they appear to already operate differently. Investigating:

## Git

```powershell
PS C:\Users\andrew\src\test> ls


    Directory: C:\Users\andrew\src\test


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        9/22/2016  10:42 AM              0 foo
-a----        9/22/2016  10:42 AM              0 foobar
-a----        9/22/2016  10:42 AM              0 foobaz
-a----        9/22/2016  10:42 AM              0 foobil
-a----        9/22/2016  10:42 AM              0 foocod


PS C:\Users\andrew\src\test> git add foo*
PS C:\Users\andrew\src\test> git status
On branch master

Initial commit

Changes to be committed:
  (use "git rm --cached <file>..." to unstage)

        new file:   foo
        new file:   foobar
        new file:   foobaz
        new file:   foobil
        new file:   foocod
```

It did the same with PowerShell on Linux without these changes; so it globs itself on both platforms. That said, Bash definitely glob expands and it works just fine.

## GCC

On Linux without these changes:

```powershell
~/test |-/ ls                                                                                                                                             
foobar.c  foobaz.c  foobil.c  foo.c  foocod.c
~/test |-/ gcc *.c                                                                                                                                        
gcc: error: *.c: No such file or directory
gcc: fatal error: no input files
compilation terminated.
```

But with GCC through MinGW (i.e. *not Cygwin*) on Windows, GCC globs:

```powershell
PS C:\Users\andrew\src\test> ls


    Directory: C:\Users\andrew\src\test


Mode                LastWriteTime         Length Name
----                -------------         ------ ----
-a----        9/22/2016  11:07 AM              0 foo.c
-a----        9/22/2016  11:07 AM              0 foobar.c
-a----        9/22/2016  11:07 AM              0 foobaz.c
-a----        9/22/2016  11:07 AM              0 foobil.c
-a----        9/22/2016  11:07 AM              0 foocod.c


PS C:\Users\andrew\src\test> C:\MinGW\bin\gcc.exe *.c
PS C:\Users\andrew\src\test>
PS C:\Users\andrew\src\test> C:\MinGW\bin\gcc.exe *.blah
gcc.exe: error: *.blah: Invalid argument
gcc.exe: fatal error: no input files
compilation terminated.
```

We can't test everything, but so far @lzybkr appears right, and tools built for Windows (even cross-platform ones like Git and GCC) don't expect globbing to be done for them, and so do it themselves.